services/queue/src/main/java/tech/pegasys/pantheon/services/queue/RocksDbTaskQueue.java

```diff
@@ -112,9 +112,25 @@ public synchronized long size() {
 
   @Override
   public synchronized boolean isEmpty() {
+    assertNotClosed();
     return size() == 0;
   }
 
+  @Override
+  public synchronized void clear() {
+    assertNotClosed();
+    outstandingTasks.clear();
+    byte[] from = Longs.toByteArray(oldestKey.get());
+    byte[] to = Longs.toByteArray(lastEnqueuedKey.get() + 1);
+    try {
+      db.deleteRange(from, to);
+      lastDequeuedKey.set(0);
+      lastEnqueuedKey.set(0);
```

Comment:
Do we need to set `oldestKey` to 0 as well?

Reply:
yes - good catch.